src/pages/index.js

```diff
@@ -54,31 +91,216 @@ variables.profileEditButton.addEventListener("click", () => {
+/* -------------------------------------------------------------------------- */
+
+addCardPopup.setEventListeners();
 variables.addButton.addEventListener("click", () => {
-  addFormPopup.open();
+  addCardPopup.open();
 });
-//
-const cardSection = new Section(
-  {
-    items: data,
-    renderer: renderCard,
-  },
-  "#el-card-list"
+
+/* -------------------------------------------------------------------------- */
+/*                  Pass form inputs to validate *Avatar* form data           */
+/* -------------------------------------------------------------------------- */
+
+const avatarFormValidator = new FormValidator(
+  formConfig,
+  variables.avatarModalForm
+);
+avatarFormValidator.enableValidation();
+
+/* -------------------------------------------------------------------------- */
+/*                         Call API and pass info to update the avatar        */
+/* -------------------------------------------------------------------------- */
+
+const avatarPopupForm = new PopupWithForm(
+  "#edit-avatar",
+  ({ link: avatar }) => {
+    variables.avatarModalSaveButton.textContent = "Saving...";
+    api
+      .updateUserAvatar({ avatar })
+      .then((res) => {
+        user.updateAvatar(res.avatar);
+        avatarPopupForm.close();
+        variables.avatarModalForm.reset();
+      })
+      .catch((error) => {
+        console.error("Failed to update user avatar:", error);
+      })
+      .finally(() => {
+        variables.avatarModalSaveButton.textContent = "Save";
+        avatarFormValidator.disableSubmitButton();
+      });
+  }
 );
 
-function renderCard(item) {
-  const cardElement = createCard(item);
-  cardSection.addItem(cardElement);
+/* -------------------------------------------------------------------------- */
+/*                   Edit Avatar Form Event Listener                          */
+/* -------------------------------------------------------------------------- */
+
+avatarPopupForm.setEventListeners();
+variables.avatarChangeBut.addEventListener("click", () => {
+  avatarPopupForm.open();
+});
+
+/* -------------------------------------------------------------------------- */
+/*                  Create card object and render the images                  */
+/* -------------------------------------------------------------------------- */
+
+function createCard(card) {
+  const newCard = new Card(
+    card,
+    "#elementCard",
+    handleImageClick,
+    handleDeleteCardButtonClick,
+    likeCardAPI,
+    unlikeCardAPI
+  );
+
+  return newCard.getView();
 }
 
-cardSection.renderItems();
+/* -------------------------------------------------------------------------- */
+/*              Handle like/dislike functionality callback                    */
+/* -------------------------------------------------------------------------- */
+
+const likeCardAPI = (cardData, changeLike) => {
+  api
+    .likeCard(cardData)
+    .then(() => {
+      changeLike.classList.toggle("element__like-button_active");
```

Comment:
**Could be improved**

To have access to any method of the class in `index.js` you can pass `this` to the call:

```js

        this._likeButton.addEventListener('click', () => {
            this._handleLike(this);
        });


```

And now in `index.js` you can call the methods like this:


```js


function addLike(cardInstance) {
        api.likeCard(cardInstance.getId())
            .then((res) => {
               cardInstance.toggleLike()
            })
            .catch((err) => {
                console.error(err);
            });
}


```

The same thing is with deleting cards